src/living/helpers/gui2d/control.ts

```diff
@@ -739,9 +756,71 @@ export class Control2D {
       element.width = rect.width;
       element.height = rect.height;
     });
-    renderingContext.putImageData(this._imageData, rect.x, rect.y, 0, 0, rect.width, rect.height);
+    renderingContext.drawImage(this._imageBitmap, rect.x, rect.y, rect.width, rect.height);
   }
 
+  private _updateCurrentTransformMatrix() {
+    const element = this._element;
+    const style = this._style;
+    if (element instanceof ShadowRootImpl) {
+      return;
+    } 
+    const transformStr = style.transform;
+    const parentElement = element.parentElement;
+    const transforms = parseTransform(transformStr)
+    this.currentTransformMatrix = this._calculateTransformMatrix(transforms);
+    if (parentElement === null) {
```

Comment:
```suggestion
    if (parentElement == null) {
```
Use non-strict equal for null check.